src/actions/index.jsx

```diff
@@ -1,12 +1,43 @@
 import { auth, provider, storage } from "../firebase";
 import db from "../firebase";
+
 import {
   SET_DARK_MODE,
   SET_USER,
   SET_LOADING_STATUS,
   GET_ARTICLES,
 } from "./actionType";
 
+export function generateKeyWords(name, email) {
+  const words = [];
+
+  name.split(" ").forEach((word) => words.push(word));
+  words.push(email.split("@")[0]);
+
+  return words;
+}
+
+export async function setFbUser(userData) {
+  const userQuery = db.collection("users").doc(userData.email);
+
+  const data = {
+    connections: [],
+    keywords: generateKeyWords(userData.displayName, userData.email),
+    name: userData.displayName,
```

Comment:
Is this information being set here or just recorded? If it is being set here, I'll need to update this data when the user edits their profile. We really should round up all the db. collections in the app and put together a visual schema. I can do it after we mostly write the these next 3 pages because I'm already working on something similar for project reclass so it will all still be fresh in my head. Then we can update them where ever for maximum efficiency.